### DISCUSSION TITLE
id like to use Frigate as my main nvr- how to change path for saving clips?

### DISCUSSION THREAD

**Discussion by doronazl** [blakeblackshear/frigate#1097] [Ask A Question]
i bought an extra hdd to dedicate for the clips
i cant find a way to change the path for saving the clips
i use SAMBA NAS addon to expose my hdd to HA OS , it shows up, how do i tell frigate to save the clips there ?

**Comment by blakeblackshear** [comment]
I don't think anyone has figured this out when using the addon. Have you been able to use it successfully with other addons? If so, can you provide more details on how?

**Reply by doronazl** [reply]
Yes i have done it with motioneye.
I used samba nas addon to expose the hdd and afterwards tried manually entering the path in motioneye file storage menu which did not work for some reason, so i then used the network share option in motioneye instead of path,
You enter the ip of the ha server, choose smb protocol 3.1.1 user and pass and the root path and thats it.
It recognized the capacity and everything. 

 how do people work otherwise? Just use the main hdd as storage for the clips?

**Reply by blakeblackshear** [reply]
So the network drive stuff is built into motioneye then. I can look to see how that works. Many people run frigate without clips or recordings or just use the main hard drive with a shorter retention.

**Reply by doronazl** [reply]
So basically most using the object tracking to get triggers for automations?
Not as full NVR?

**Reply by blakeblackshear** [reply]
Many people are not running as an addon or on a separate host so they can store clips wherever you want.

**Reply by mechpanos** [reply]
Why don't you try linux symlink capabilities? You can symlink the hard drive to the folder where frigate saves clips and snapshots. The tricky thing is that, it depends on where you run homeassistant on: proxmox, HASSOS, Debian/Other OS, etc, if you are able to do it, or not...

**Reply by HIT2022** [reply]
I'm guessing if he is on HA OS or HA Container, he doesn't have an underlying linux OS to access. I am on Supervised, so for me this requirement is a simple matter of mounting the NAS to /usr/share/hassio/media, which the add-on (docker container) then inherits at run-time

**Comment by doronazl** [comment]
I see, do u think this is something can be fixed anytime soon? Or should
explore other options?

**Reply by blakeblackshear** [reply]
It would be easy to transition back to the addon when it is resolved. Not sure how long it will take. If you can run directly via portainer or docker, that may be your best option for now.

**Reply by fullhack2** [reply]
Hello all,

Just wondering if there has been any movement on adding the ability to add an SMB mount and adjust configuration within the HA addon, or if there is a suggestion on how to continue to use the addon in HA while redirecting the 'frigate' folder created within 'media' to an external location?  I do see hunterjm's response below on redirecting the entire data partition, but wasn't sure how it would/could work with a share and not a local disk.

Thanks!

**Reply by blakeblackshear** [reply]
I think the best way is to run frigate outside of home assistant and use the proxy addon. Do you have a way to run it directly in docker?

**Reply by hunterjm** [reply]
I am running the Addon, but on Supervised instead of OS. At that point it's just a matter of creating a mount in `fstab` pointing to the frigate folder in media which makes the OS think it's writing locally to the default location, but is really writing to the network mount. I don't think that's possible in OS, and you should setup Frigate on a separate computer as @blakeblackshear recommended.

**Reply by fullhack2** [reply]
Thanks for the suggestions.

I should be able to set up a separate docker instance and run it outside of HA.  I just really liked the consolidated HA environment and having Frigate as an addon was nice (just backing up my HA instance, etc. without having separate components to worry about).  I was previously using MotionEye and I know it had the option to mount SMB shares directly from the app config, which was super convenient.

**Reply by fullhack2** [reply]
ok, so i set up a test Docker instance to sort this out independently of HA.  I managed to get the redirect to SMB mount but ran into a permission issue (snapshots and clips aren't presented in the web interface due to invalid permissions).  This is probably really simple, but I'm feeling dumb at the moment:

**nginx error log on container:**
021/09/08 01:44:45 [error] 20#20: *518 open() "/media/frigate/clips/bonus_room_cam-1631065319.593839-8wam8p.mp4" failed (13: Permission denied), client: 192.168.1.111, server: , request: "GET /clips/bonus_room_cam-1631065319.593839-8wam8p.mp4 HTTP/1.1", host: "192.168.1.216:5000", referrer: "http://192.168.1.216:5000/events/1631065319.593839-8wam8p"

**file permissions as seen by the container:**
root@a038cc430862:/media/frigate/clips# ls -al
total 27632
drwxrwx--- 1 root 998    4096 Sep  8 01:51 .
drwxrwx--- 1 root 998       0 Sep  8 01:39 ..
-rwxrwx--- 1 root 998  197804 Sep  8 01:40 bonus_room_cam-1631065193.800552-qojobo.jpg
-rwxrwx--- 1 root 998 2510556 Sep  8 01:40 bonus_room_cam-1631065193.800552-qojobo.mp4
-rwxrwx--- 1 root 998  196701 Sep  8 01:41 bonus_room_cam-1631065268.414669-99u60y.jpg
-rwxrwx--- 1 root 998 1928727 Sep  8 01:41 bonus_room_cam-1631065268.414669-99u60y.mp4
-rwxrwx--- 1 root 998  201233 Sep  8 01:42 bonus_room_cam-1631065319.593839-8wam8p.jpg
-rwxrwx--- 1 root 998 3070112 Sep  8 01:42 bonus_room_cam-1631065319.593839-8wam8p.mp4
-rwxrwx--- 1 root 998  204538 Sep  8 01:42 bonus_room_cam-1631065366.360399-cfrj2z.jpg
-rwxrwx--- 1 root 998 2084833 Sep  8 01:43 bonus_room_cam-1631065366.360399-cfrj2z.mp4
-rwxrwx--- 1 root 998  193731 Sep  8 01:46 bonus_room_cam-1631065560.524629-thd687.jpg
-rwxrwx--- 1 root 998 2165569 Sep  8 01:46 bonus_room_cam-1631065560.524629-thd687.mp4
-rwxrwx--- 1 root 998  194668 Sep  8 01:46 bonus_room_cam-1631065568.187359-21593z.jpg
-rwxrwx--- 1 root 998 2110829 Sep  8 01:46 bonus_room_cam-1631065568.187359-21593z.mp4
-rwxrwx--- 1 root 998   32768 Sep  8 01:39 frigate.db
-rwxrwx--- 1 root 998   32768 Sep  8 01:39 frigate.db-shm
-rwxrwx--- 1 root 998  498552 Sep  8 01:51 frigate.db-wal
-rwxrwx--- 1 root 998  195635 Sep  8 01:39 front_cam-1631065160.754096-22o537.jpg
-rwxrwx--- 1 root 998 3477672 Sep  8 01:39 front_cam-1631065160.754096-22o537.mp4
-rwxrwx--- 1 root 998  194135 Sep  8 01:39 front_cam-1631065171.161327-n5v06h.jpg
-rwxrwx--- 1 root 998 2043463 Sep  8 01:39 front_cam-1631065171.161327-n5v06h.mp4
-rwxrwx--- 1 root 998  257189 Sep  8 01:39 garage_cam-1631065160.658395-5lxy32.jpg
-rwxrwx--- 1 root 998 2494675 Sep  8 01:39 garage_cam-1631065160.658395-5lxy32.mp4
-rwxrwx--- 1 root 998  257048 Sep  8 01:39 garage_cam-1631065170.405517-el71yi.jpg
-rwxrwx--- 1 root 998 1571628 Sep  8 01:39 garage_cam-1631065170.405517-el71yi.mp4
-rwxrwx--- 1 root 998  253430 Sep  8 01:51 garage_cam-1631065850.928494-4o9ezk.jpg
-rwxrwx--- 1 root 998 1859295 Sep  8 01:51 garage_cam-1631065850.928494-4o9ezk.mp4

Any thoughts?  Like I said this is probably really simple and I'm just overlooking something.

**Reply by fullhack2** [reply]
figured it out I believe.  As I'm using VirtualBox and have the mount being done through their 'Guest Additions' I had to add 'root' to the 'vboxsf' group and restart.  Not sure if this is the best way, but it works for the time being.

**Reply by fullhack2** [reply]
just kidding... thought I fixed it, but it somehow reverted back to the local /media/frigate and ignored the mounted drive when I restarted the whole VM (which may be another issue and I hope doesn't occur each time I reboot).  I bounced the container and then it mounted properly, but with the same results (web interface cannot load files).  The mount shows as follows in the container:

D_DRIVE on /media/frigate type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=0,gid=998,dmode=0770,fmode=0770,tag=VBoxAutomounter)

Any help would be appreciated.

**Reply by hunterjm** [reply]
Your setup seems overly complex. What is the point in running Docker (a containerization environment) inside a virtual machine? You are not likely to find many people with your same configuration, so it will be hard for others to help troubleshoot your problems.

**Reply by fullhack2** [reply]
In my travels I have found a lot of people that run their Docker environments on VMs, especially when they don't have the luxury of a dedicated host, etc. so I don't think this is an overly complex setup by any means.  I believe by design, Docker is to allow you the flexibility to deploy anywhere, regardless of the underlying architecture/setup (physical host, nested virtualization, etc.).

I did manage to figure out the permissions issue (added a group 'vboxsf' to GID 998 in my case, and then added 'nobody' to the 'vboxsf' group to match the filesystem on the Docker Host).  I would just like to figure out the best way to add the group creation and user addition to the docker-compose script for future ease of deployment.

**Comment by doronazl** [comment]
Never heard of it, can u do it on HA OS?

**Comment by hunterjm** [comment]
On HA OS you would have to move the entire data partition to the other hard drive. This removes the need for the samba NAS addon entirely. Documentation here: https://github.com/home-assistant/operating-system/blob/master/Documentation/partition.md#using-datactl-to-move-the-data-partition

**Comment by blakeblackshear** [comment]
Someone found a way to get HassOS to mount a different disk in the media folder: https://community.home-assistant.io/t/solved-mount-usb-drive-in-hassio-to-be-used-on-the-media-folder-with-udev-customization/258406/46?u=blakeblackshear
